open-amp/lib/utils/string.c

```diff
@@ -49,5 +50,5 @@ strlcpy(char *dst, const char *src, size_t size)
 	if (nleft == 0 && size != 0)
 		*dst = '\0';		/* NUL-terminate dst */
 
-	return strlen(src);
+	return (s - src - 1);
```

Comment:
The function has to return  the  total  length  of the  string  it tried to create so the size of the source (https://man.freebsd.org/cgi/man.cgi?strlcpy(3))
The freebsd add in code `while (*src++)` resulting in same overflow issue than strlen, but perhaps not...
if you detect a potential issue, please, comment https://github.com/OpenAMP/open-amp/pull/620 with your observation that I fix it in the openamp repo